test/e2e/query_tickets_test.go

```diff
@@ -17,282 +17,145 @@
-						},
-					},
-				},
-			},
-			wantPageCount: 1,
-		},
+func TestNoPool(t *testing.T) {
+	om, closer := e2e.New(t)
+	defer closer()
+
+	q := om.MustQueryServiceGRPC()
+
+	stream, err := q.QueryTickets(context.Background(), &pb.QueryTicketsRequest{Pool: nil})
+	assert.Nil(t, err)
+
+	resp, err := stream.Recv()
+	assert.Equal(t, codes.InvalidArgument, status.Convert(err).Code())
+	assert.Nil(t, resp)
+}
+
+func TestNoTickets(t *testing.T) {
+	om, closer := e2e.New(t)
+	defer closer()
+
+	q := om.MustQueryServiceGRPC()
+	stream, err := q.QueryTickets(context.Background(), &pb.QueryTicketsRequest{Pool: &pb.Pool{}})
+	assert.Nil(t, err)
+
+	resp, err := stream.Recv()
+	assert.Equal(t, io.EOF, err)
+	assert.Nil(t, resp)
+}
+
+func TestPaging(t *testing.T) {
+	om, closer := e2e.New(t)
+	defer closer()
+
+	pageSize := 10 // TODO: read from config
+	if pageSize < 1 {
+		assert.Fail(t, "invalid page size")
 	}
 
-	t.Run("TestQueryTickets", func(t *testing.T) {
-		for _, test := range tests {
-			test := test
-			t.Run(test.description, func(t *testing.T) {
-				t.Parallel()
-
-				om, closer := e2e.New(t)
-				defer closer()
-				fe := om.MustFrontendGRPC()
-				mml := om.MustQueryServiceGRPC()
-				pageCounts := 0
-				ctx := om.Context()
-
-				for _, ticket := range test.gotTickets {
-					resp, err := fe.CreateTicket(ctx, &pb.CreateTicketRequest{Ticket: ticket})
-					assert.NotNil(t, resp)
-					assert.Nil(t, err)
-				}
-
-				stream, err := mml.QueryTickets(ctx, &pb.QueryTicketsRequest{Pool: test.pool})
-				assert.Nil(t, err)
-
-				var actualTickets []*pb.Ticket
-
-				for {
-					resp, err := stream.Recv()
-					if err == io.EOF {
-						break
-					}
-					if err != nil {
-						assert.Equal(t, test.wantCode, status.Convert(err).Code())
-						break
-					}
-
-					actualTickets = append(actualTickets, resp.Tickets...)
-					pageCounts++
-				}
-
-				require.Equal(t, len(test.wantTickets), len(actualTickets))
-				// Test fields by fields because of the randomness of the ticket ids...
-				// TODO: this makes testing overcomplicated. Should figure out a way to avoid the randomness
-				// This for loop also relies on the fact that redis range query and the ticket generator both returns tickets in sorted order.
-				// If this fact changes, we might need an ugly nested for loop to do the validness checks.
-				for i := 0; i < len(actualTickets); i++ {
-					assert.Equal(t, test.wantTickets[i].GetAssignment(), actualTickets[i].GetAssignment())
-					assert.Equal(t, test.wantTickets[i].GetSearchFields(), actualTickets[i].GetSearchFields())
-				}
-				assert.Equal(t, test.wantPageCount, pageCounts)
-			})
+	totalTickets := pageSize*5 + 1
+	expectedIds := map[string]struct{}{}
+
+	fe := om.MustFrontendGRPC()
+	for i := 0; i < totalTickets; i++ {
+		resp, err := fe.CreateTicket(context.Background(), &pb.CreateTicketRequest{Ticket: &pb.Ticket{}})
+		assert.NotNil(t, resp)
+		assert.NotNil(t, resp.Ticket)
+		assert.Nil(t, err)
```

Comment:
use require package instead? assert wont exit the test flow, if CreateTicket failed then the rest of the assertions are meaningless.

Reply:
I didn't want to pull in a different package and make the choice of assert or require for each call.  It will fail anyways?  Does it make a difference?

Reply:
the difference is it will fail later with a panic without using require

Reply:
That seems fine?  It's still failing.

Reply:
the drawback of this approach is you would likely lost in the errors and couldn't find where the test began failing ಠ_ಠ... anyway I don't have a strong opinion about this, this is just a nit.